docs/learn/smart-contract-developers/create-nft-collection-with-lsp7-tokenId.md

```diff
@@ -0,0 +1,953 @@
+
+```json
+{
+  "LSP4Metadata": {
+    "name": "The Dematerialised",
+    "description": "The Experiential Marketspace For Digital Goods",
+    "links": [
+      {
+        "title": "Website",
+        "url": "https://thedematerialised.com"
+      }
+    ],
+    "icons": [],
+    "images": [
+      {
+        "width": 1024,
+        "height": 974,
+        "url": "ipfs://QmS3jF9jsoG6gnyJ7wCeJ4bej2aJEnPSv527UV8KxjBDAA",
+        "verification": {
+          "method": "keccak256(bytes)",
+          "data": "0xdd6b5fb6dc984fda0222fb6f6e96b471c0667b12f03b1e804f7b5e6ab62acdb0"
+        }
+      }
+    ],
+    "assets": [],
+    "attributes": []
+  }
+}
+```
+
+We will do the same with the LSP7 token metadata. Let's create a `lsp7TokenMetadata.json` file in the `metadata` folder and set the following metadata:
+
+```json
+{
+  "LSP4Metadata": {
+    "name": "KLxENDLESS MEDALLION Purple",
+    "description": "Collaboration with Karl Largerfeld",
+    "links": [
+      {
+        "title": "Get yours now",
+        "url": "https://thedematerialised.com/shop/karl-lagerfeld/klxendless-medallion"
+      }
+    ],
+    "icons": [
+      {
+        "width": 256,
+        "height": 256,
+        "url": "ipfs://QmS3jF9jsoG6gnyJ7wCeJ4bej2aJEnPSv527UV8KxjBDAA",
+        "verification": {
+          "method": "keccak256(bytes)",
+          "data": "0xdd6b5fb6dc984fda0222fb6f6e96b471c0667b12f03b1e804f7b5e6ab62acdb0"
+        }
+      }
+    ],
+    "images": [
+      [
+        {
+          "width": 1024,
+          "height": 974,
+          "url": "ipfs://QmUGmycxrwFec15UC41v9bvnRStK3zxR7mth72mGRcUSPD",
+          "verification": {
+            "method": "keccak256(bytes)",
+            "data": "0x951bf983a4b7bcebc5c0b00a5e783630dcb788e95ee9e44b0b7d4bde4a0b4d81"
+          }
+        }
+      ]
+    ],
+    "assets": [
+      {
+        "verification": {
+          "method": "keccak256(bytes)",
+          "data": "0x88f3d704f3d534267c564019ce2b70a5733d070e71bf2c1f85b5fc487f47a46f"
+        },
+        "url": "ifps://QmTDQGR26dSd3c4qJpmFwTh7gNRPnNbBf2Fg3gULypUag3",
+        "fileType": "mp4"
+      }
+    ],
+    "attributes": []
+  }
+}
+```
+
+### Deploy the LSP8 token script
+
+Let's create a `deployLSP8Token.ts` file in the `scripts` folder.
+
+#### Imports
+
+For this script we will jut need to import:
+
+- the `ethers` library to interact with the blockchain
+- the ERC725 library to convert the metadata to [VerifiableURI](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#verifiableuri)
+- the LSP8 token metadata json file we just created
+- the LSP4 metadata schema
+
+```typescript
+import { ethers } from 'hardhat';
+import { ERC725, ERC725JSONSchema } from '@erc725/erc725.js';
+import lsp8TokenMetadata from './metadata/lsp8TokenMetadata.json';
+import LSP4DigitalAsset from '@erc725/erc725.js/schemas/LSP4DigitalAsset.json';
```

Comment:
I suggest to order the imports

```suggestion
import { ethers } from 'hardhat';

import { ERC725, ERC725JSONSchema } from '@erc725/erc725.js';
import LSP4DigitalAsset from '@erc725/erc725.js/schemas/LSP4DigitalAsset.json';

import lsp8TokenMetadata from './metadata/lsp8TokenMetadata.json';
```

+ if i follow the guide up to here I get

You need to ask user to install erc725js with npm

<img width="889" alt="image" src="https://github.com/lukso-network/docs/assets/477945/50f104ff-172b-43ee-ab3d-97b260d3a390">

Reply:
weird because i added the it to eh prerequisite section